scalding-core/src/main/scala/com/twitter/scalding/ExecutionUtil.scala

```diff
@@ -0,0 +1,92 @@
+package com.twitter.scalding
+
+import java.util.concurrent.Semaphore
+
+import com.twitter.algebird.Monoid
+import scala.util.{ Failure, Success }
+
+object ExecutionUtil {
+  /**
+   * Run a sequence of executions but only permitting parallelism amount to run at the
+   * same time.
+   *
+   * @param executions List of executions to run
+   * @param parallelism Number to run in parallel
+   * @return Execution Seq
+   */
+  def withParallelism[T](executions: Seq[Execution[T]], parallelism: Int): Execution[Seq[T]] = {
+    require(parallelism > 0, s"Parallelism must be > 0: $parallelism")
+
+    val lock = new Semaphore(parallelism)
+
+    def waitRun(e: Execution[T]): Execution[T] = {
+      Execution.from(lock.acquire())
```

Comment:
That deserves a good test, the Execution.from implementation is such I'm not sure if it does, there is a lot of lazy creation of things that its possible its deferred into a future pool. -- However its still not great since it will block a thread in the future pool rather than behaving like a Promise would